server/src/main/java/org/opensearch/index/store/remote/metadata/RemoteSegmentMetadata.java

```diff
@@ -99,19 +103,57 @@ public static Map<String, RemoteSegmentStoreDirectory.UploadedSegmentMetadata> f
 
     public void write(IndexOutput out) throws IOException {
         out.writeMapOfStrings(toMapOfStrings());
-        out.writeLong(generation);
-        out.writeLong(primaryTerm);
+        writeCheckpointToIndexOutput(replicationCheckpoint, out);
         out.writeLong(segmentInfosBytes.length);
         out.writeBytes(segmentInfosBytes, segmentInfosBytes.length);
     }
 
     public static RemoteSegmentMetadata read(IndexInput indexInput) throws IOException {
         Map<String, String> metadata = indexInput.readMapOfStrings();
-        long generation = indexInput.readLong();
-        long primaryTerm = indexInput.readLong();
+        ReplicationCheckpoint replicationCheckpoint = readCheckpointFromIndexInput(indexInput);
         int byteArraySize = (int) indexInput.readLong();
         byte[] segmentInfosBytes = new byte[byteArraySize];
         indexInput.readBytes(segmentInfosBytes, 0, byteArraySize);
-        return new RemoteSegmentMetadata(RemoteSegmentMetadata.fromMapOfStrings(metadata), segmentInfosBytes, primaryTerm, generation);
+        return new RemoteSegmentMetadata(RemoteSegmentMetadata.fromMapOfStrings(metadata), segmentInfosBytes, replicationCheckpoint);
+    }
+
+    public static void writeCheckpointToIndexOutput(ReplicationCheckpoint replicationCheckpoint, IndexOutput out) throws IOException {
+        writeShardIdToIndexOutput(replicationCheckpoint.getShardId(), out);
```

Comment:
can we read/write write the shard fields here directly and avoid all these static calls?

```java
        out.writeString(index.getName());
        out.writeString(index.getUUID());
        out.writeVInt(shardId.getId());
```
```java
new ShardId(new Index(in.readString(), in.readString(), in.readVInt());
```

```

Reply:
Updated.